core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -44,7 +44,7 @@ os_mmap(void *hint, size_t size, int prot, int flags)
     page_size = (uint64)getpagesize();
     request_size = (size + page_size - 1) & ~(page_size - 1);
 
-#if !defined(__APPLE__) && !defined(__NuttX__)
+#if !defined(__APPLE__) && !defined(__NuttX__) && defined(MADV_HUGEPAGE)
```

Comment:
Should also modify other places, including L19, L151, and L203, for L203, could you please help change it to code like `/* end of !defined(__APPLE__) && !defined(__NuttX__) && defined(MADV_HUGEPAGE) */`?

Reply:
I modified it. Please review it.